swap_meet/vendor.py

```diff
@@ -1,2 +1,82 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        if their_item not in vendor.inventory or my_item not in self.inventory:
+            return False
+        else:
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+            return True
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) == 0 or len(vendor.inventory) == 0:
+            return False
+        else:
+            my_new_item = vendor.inventory[0]
+            their_new_item = self.inventory[0]
+            self.swap_items(vendor, their_new_item, my_new_item)
+            return True
+
+    def get_best_by_category(self, category):
+        best_quality = 0
+        try:
+            for item in self.inventory:
+                if item.category == category:
+                    if item.condition > best_quality:
+                        best_quality = item.condition
```

Comment:
Notice that a significant portion of this logic is filtering the inventory by a certain category. Don't we have a function that returns a filtered view of the inventory down to some category?